src/main/java/com/google/api/codegen/discogapic/MainDiscoGapicProviderFactory.java

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2017 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discogapic;
+
+import com.google.api.codegen.config.GapicProductConfig;
+import com.google.api.codegen.config.PackageMetadataConfig;
+import com.google.api.codegen.discogapic.transformer.java.JavaDiscoGapicSurfaceTransformer;
+import com.google.api.codegen.discovery.Document;
+import com.google.api.codegen.gapic.CommonGapicCodePathMapper;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.gapic.GapicGeneratorConfig;
+import com.google.api.codegen.rendering.CommonSnippetSetRunner;
+import com.google.api.codegen.util.java.JavaRenderingUtil;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.NotImplementedException;
+
+public class MainDiscoGapicProviderFactory implements DiscoGapicProviderFactory {
```

Comment:
"Main" does not seem very informative. What other implementations of `DiscoGapicProviderFactory` are envisioned?